utils.ts

```diff
@@ -9,26 +9,60 @@ import {join} from 'path';
 import {execSync} from 'child_process';
 import * as mkdirp from 'mkdirp';
 
-export function defaults<T>(val: T | undefined, def: T): T {
+export function defaults<T>(val: T|undefined, def: T): T {
   return typeof val === 'undefined' ? def : val;
 }
 
 export async function delay(time: number) {
   return new Promise(resolve => setTimeout(resolve, time));
 }
 
+export function getPlatform() {
+  const platform = process.platform;
+
+  if (platform !== 'linux') {
+    return platform;
+  }
+  // Windows Subsystem for Linux, reports Linux as platform but chrome is on Windows
+  // we need to find out the real platform
+  if (process.env.PATH && (process.env.PATH as string).includes(':/mnt/c/Windows/System32:')) {
+    return 'wsl';
+  }
+
+  return platform;
+}
+
 export function makeTmpDir() {
-  switch (process.platform) {
+  switch (getPlatform()) {
     case 'darwin':
     case 'linux':
       return makeUnixTmpDir();
+    case 'wsl':
+      // We populate the user's Windows temp dir so the folder is correctly created later
+      process.env.TEMP = `/mnt/c/Users/${findWSLUser()}/AppData/Local/Temp`;
```

Comment:
This isn't set for WSL typically ?  Separately, I imagine `TMPDIR` is set, yeah? Can we use that'?

Reply:
Unfortunately it isn't set. Also because we are on Linux the tempdir is different than the Windows one. and Chrome (from Windows) can't access the WSL tempdir.
RE the user name, it can be different between Windows and Linux. I spent a few minutes trying to figure out what was wrong on my machine because of that :(